manager/state/raft/raft.go

```diff
@@ -270,6 +275,9 @@ func (n *Node) JoinAndStart() error {
 			n.Node = raft.StartNode(n.Config, []raft.Peer{})
 
 			if err := n.registerNodes(resp.Members); err != nil {
```

Comment:
It will be caught by the defer because it's a named return, so returning the value sets the function-scoped `err` regardless of what happened before.